File: languages/java8/template/src/main/java/com/algorithmia/algorithm/Algorithm.java
package com.algorithmia.algorithm;

import com.algorithmia.development.*;
import com.algorithmia.*;

class Algorithm implements AlgorithmInterface<String, String>{


    public String apply(String input){
        return "hello ".concat(input);
    }

    public void load(){
        System.out.println("Implement this for loading functionality");
    }

    public static void main(String[] args) {
        Algorithm algorithm = new Algorithm();
        Handler algo = new Handler<>(algorithm);
        algo.serve();
    }
}

File: languages/scala-2/template/src/main/scala/com/algorithmia/Algorithm.scala
package com.algorithmia

import com.algorithmia.handler.AbstractAlgorithm

import scala.util.{Success, Try}

class Algorithm extends AbstractAlgorithm[String, String]{

  var someVariable: String = _

  override def apply(input: String): Try[String] = {
    Success(s"hello $input")
  }
  override def load = {
    someVariable = "loaded"
    Success()
  }
}

object Algorithm {
  val handler = Algorithmia.handler(new Algorithm)

  def main(args: Array[String]): Unit = {
    handler.serve()
  }
}
